### PR TITLE
Service to service troubleshooting

### PR DESCRIPTION
**Pull request by curtbushko**
Things to still do:

- Clean up text strings based on designer input
- Add acceptance test when troubleshoot package is working with validation
- Remove hardcoded replace in go.mod when branches are merged into the consul repo

Changes proposed in this PR:
-
-

How I've tested this PR:

How I expect reviewers to test this PR:

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Review by ndhanushkodi — COMMENTED** [review]
Amazing!! Left a few comments!

This could definitely come in a future PR, but could we add a case for testing with a config entry (deploy like static-server1 and static-server2 and do a splitter between them) to the acceptance test?

File: acceptance/tests/cli/cli_install_test.go [hunk 1]
```diff
@@ -83,6 +87,41 @@ func TestInstall(t *testing.T) {
...
+			require.NoError(t, err)
+
+			clientPodName := clientPod.Items[0].Name
+			upstreamsOut, err := cli.Run(t, ctx.KubectlOptions(t), "troubleshoot", "upstreams", "-pod", clientPodName)
+			logger.Log(t, string(upstreamsOut))
+			require.NoError(t, err)
+
+			if c.tproxy {
+				// With tproxy disabled and explicit upstreams we need the upstream ip of the static-server. This is the
+				// ClusterIP of the service
```
[the remark this answers is not in the document]
Comment (curtbushko) [reply acceptance/tests/cli/cli_install_test.go:103]: Good catch!  I had moved a bunch of things around to tighten up the tests and jumbled it all!

File: acceptance/tests/cli/cli_install_test.go [hunk 2]
```diff
@@ -83,6 +87,41 @@ func TestInstall(t *testing.T) {
...
+				serverIP := serverService.Items[0].Spec.ClusterIP
+
+				proxyOut, err := cli.Run(t, ctx.KubectlOptions(t), "troubleshoot", "proxy", "-pod", clientPodName, "-upstream-ip", serverIP)
+				require.NoError(t, err)
+				require.Regexp(t, "upstream resources are valid", string(proxyOut))
+				logger.Log(t, string(proxyOut))
+			} else {
+				// If tproxy is enabled we are looking for the upstream envoy id
+				require.Regexp(t, "static-server", string(upstreamsOut))
+				logger.Log(t, string(upstreamsOut))
```
[the remark this answers is not in the document]
Comment (curtbushko) [reply acceptance/tests/cli/cli_install_test.go:117]: Removed

File: cli/cmd/troubleshoot/proxy/proxy.go [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+	if err != nil {
+		return err
+	}
+
+	t, err := troubleshoot.NewTroubleshoot(adminAddrIP, adminPort)
+	if err != nil {
+		return err
+	}
+
+	err = t.GetEnvoyConfigDump()
```
[the remark this answers is not in the document]
Comment (curtbushko) [reply cli/cmd/troubleshoot/proxy/proxy.go:215]: Removed!

File: cli/cmd/troubleshoot/proxy/proxy.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+	return 0
+}
+
+// validateFlags ensures that the flags passed in by the can be used.
+func (c *ProxyCommand) validateFlags() error {
+
+	if c.flagUpstreamEnvoyID == "" && c.flagUpstreamIP == "" {
+		return fmt.Errorf("-upstream-envoy-id OR -upstream-ip is required.\n Please run `consul troubleshoot upstreams` to find the corresponding upstream.")
+	}
```
[the remark this answers is not in the document]
Comment (curtbushko) [reply cli/cmd/troubleshoot/proxy/proxy.go:141]: I think I have it:

None:
> $ ./cli/bin/consul-k8s troubleshoot proxy -pod static-client-5644c6f9c9-rl8kf 
>                                                                                                                                       
>  ! Invalid argument: -upstream-envoy-id OR -upstream-ip is required.
>  Please run `consul troubleshoot upstreams`  to find the corresponding upstream.

Both:
> $ ./cli/bin/consul-k8s troubleshoot proxy -pod static-client-5644c6f9c9-rl8kf -upstream-ip 10.96.217.21 -upstream-envoy-id 1234
>  ! Invalid argument: -upstream-envoy-id OR -upstream-ip is required.
>  Please run `consul troubleshoot upstreams` to find the corresponding upstream.

File: cli/cmd/troubleshoot/proxy/proxy_test.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+			args: []string{},
+			out:  1,
+		},
+		"Nonexistent flag passed, -foo bar, should fail": {
+			args: []string{"-foo", "bar"},
+			out:  1,
+		},
+		"Invalid argument passed, -namespace notaname, should fail": {
+			args: []string{"-namespace", "notaname"},
+			out:  1,
```
[the remark this answers is not in the document]
Comment (curtbushko) [reply cli/cmd/troubleshoot/proxy/proxy_test.go:32]: I added test cases for setting both or none.

**Review by ndhanushkodi — APPROVED** [review]
Amazing!!!

**Pull request merged**